cluster/kubernetes/namespacer.go

```diff
@@ -68,7 +70,24 @@ func (n *namespaceViaDiscovery) EffectiveNamespace(m kresource.KubeManifest) (st
 	return m.GetNamespace(), nil
 }
 
-func (n *namespaceViaDiscovery) lookupNamespaced(groupVersion, kind string) (bool, error) {
+func (n *namespaceViaDiscovery) lookupNamespaced(groupVersion string, kind string, knownScopes ResourceDefinitionScopes) (bool, error) {
+	namespaced, clusterErr := n.lookupNamespacedInCluster(groupVersion, kind)
+	if clusterErr == nil || knownScopes == nil {
```

Comment:
If `clusterErr != nil`, you'd usually consider `namespaced` to be a suspect value.

Reply:
Not to mention, if `clusterErr != nil` and `knownScopes` is empty, the result should be an error signalling the scope cannot be determined.

Reply:
Yep, I should be returning err. Good catch, I forgot to restore that after a refactoring.

Reply:
Addressed